src/tapedfunction.jl

```diff
@@ -238,98 +218,142 @@ end
 
 
 ## Translation: CodeInfo -> Tape
-
-function bind_var!(var, bindings::Dict{Symbol, Any}, ir::Core.CodeInfo) # for literal constants
-    id = gensym()
-    bindings[id] = var
-    Box{typeof(var)}(id)
-end
-bind_var!(var::Core.SSAValue, bindings::Dict{Symbol, Any}, ir::Core.CodeInfo) =
-    bind_var!(Symbol(var.id), bindings, ir.ssavaluetypes[var.id])
-bind_var!(var::Core.TypedSlot, bindings::Dict{Symbol, Any}, ir::Core.CodeInfo) =
-    bind_var!(Symbol(:_, var.id), bindings, ir.slottypes[var.id])
-bind_var!(var::Core.SlotNumber, bindings::Dict{Symbol, Any}, ir::Core.CodeInfo) =
-    bind_var!(Symbol(:_, var.id), bindings, ir.slottypes[var.id])
-bind_var!(var::Symbol, boxes::Dict{Symbol, Any}, c::Core.Const) =
+#=
+Now, we use a Vector{Any} as the bindings to store all the
+variables used in a taped function. These variables are categorized
+into 3 kinds: 1. slot values, 2. literal data, 3. ssa values.
+
+- bindings[1:CNT_SLOT-1] is for slot values
+- bindings[CNT_SLOT] holds the current used maximum literal index
+- bindings[CNT_SLOT+1:CNT_SLOT+CNT_LITE] is for literal data
+- bindings[CNT_SLOT+CNT_LITE+1:end] is for ssa values
+=#
+const CNT_SLOT = 200
+const CNT_LITE = 1000
+const OFFSET_VAR = CNT_SLOT + CNT_LITE
+
+function bind_var!(var, bindings::Bindings, ir::Core.CodeInfo) # for literal constants
+    last_idx = bindings[CNT_SLOT]
+    idx = last_idx + 1
+    @assert idx < OFFSET_VAR
+    bindings[CNT_SLOT] = idx
+    bindings[idx] = var
```

Comment:
It is a bit unclear whether `var` is a variable name or a value. It seems that we use both in the code. It is perhaps better to avoid it to improve clarity. 
```suggestion
    bindings[idx] = var
```

Reply:
I renamed `var` to `var_literal`.